tests/interpreters/test_ml_program_interpreter.py

```diff
@@ -0,0 +1,55 @@
+from xdsl.dialects import ml_program
+from xdsl.dialects.builtin import (
+    DenseIntOrFPElementsAttr,
+    ModuleOp,
+    StringAttr,
+    SymbolRefAttr,
+    TensorType,
+    i32,
+    i64,
+)
+from xdsl.interpreter import Interpreter
+from xdsl.interpreters.ml_program import MLProgramFunctions
+from xdsl.interpreters.shaped_array import ShapedArray
+
+
+def test_ml_program_global():
+    interpreter = Interpreter(ModuleOp([]))
+    interpreter.register_implementations(MLProgramFunctions())
+    op = ml_program.Global(
+        StringAttr("global_op"),
+        TensorType(i32, [4]),
+        None,
+        DenseIntOrFPElementsAttr.tensor_from_list([4], i32, [4]),
+        StringAttr("private"),
+    )
+
+    c = interpreter.run_op(op, ())
```

Comment:
we're running into the same issue as before, I think it's time to add an assertion that `run_op` takes and returns as many interpreter values as the op has operands and results, probably in a separate PR

Reply:
The Global op is a declaration, it does not have a result, so the interpreter shouldn't either